policy/pa.go

```diff
@@ -558,14 +551,19 @@ func (pa *AuthorityImpl) ChallengeTypeEnabled(t core.AcmeChallenge) bool {
 	return pa.enabledChallenges[t]
 }
 
-// CheckAuthz determines that an authorization was fulfilled by a challenge
-// that was appropriate for the kind of identifier in the authorization.
+// CheckAuthz determines that an authorization was fulfilled by a challenge that
+// is currently enabled and was appropriate for the kind of identifier in the
+// authorization.
 func (pa *AuthorityImpl) CheckAuthz(authz *core.Authorization) error {
 	chall, err := authz.SolvedBy()
 	if err != nil {
 		return err
 	}
 
+	if !pa.ChallengeTypeEnabled(chall) {
+		return errors.New("authorization fulfilled by disabled challenge type")
```

Comment:
Since this can be seen by users we should probably be a little more specific: `authorization for %q was validated with challenge type %s, which is now disabled.`

Reply:
Would you like a similar diff a few lines down, where we say "authorization fulfilled by invalid challenge" for the case where a wildcard authz somehow has a valid HTTP-01 challenge?

Reply:
Actually in the process of writing the test I've realized that this error message is _not_ ever seen by users :(